packages/devtools_app/lib/src/flutter/extent_delegate_list.dart

```diff
@@ -0,0 +1,465 @@
+import 'dart:math' as math;
+
+import 'package:collection/collection.dart' as collection;
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+/// Base class for delegate providing extent information for items in a list.
+abstract class ExtentDelegate {
+  /// Optional callback to execute after the layout of the extents is modified.
+  VoidCallback onLayoutDirty;
+
+  int get length;
+
+  /// The main-axis extent of each item.
+  double itemExtent(int index);
+
+  /// The layout offset for the child with the given index.
+  double layoutOffset(int index);
+
+  /// The minimum child index that is visible at the given scroll offset.
+  ///
+  /// Implementations should take no more than O(log n) time.
+  int minChildIndexForScrollOffset(double scrollOffset);
+
+  /// The maximum child index that is visible at the given scroll offset.
+  ///
+  /// Implementations should take no more than O(log n) time.
+  /// Surprisingly, getMaxChildIndexForScrollOffset should be 1 less than
+  /// getMinChildIndexForScrollOffset if the scrollOffset is right at the
+  /// boundary between two items.
+  int maxChildIndexForScrollOffset(double scrollOffset);
+
+  @mustCallSuper
+  void recompute() {
+    if (onLayoutDirty != null) {
+      onLayoutDirty();
+    }
+  }
+}
+
+/// [ExtentDelegate]  implementation for the case where sizes for each
+/// item are known but absolute positions are not known.
+class FixedExtentDelegate extends ExtentDelegate {
+  FixedExtentDelegate({
+    @required this.computeExtent,
+    @required this.computeLength,
+  }) {
+    recompute();
+  }
+
+  // The _offsets list is intentionally one element larger than the length of
+  // the list as it includes an offset at the end that is the offset after all
+  // items in the list.
+  List<double> _offsets;
+
+  @override
+  int get length => _offsets.length - 1;
+
+  final double Function(int index) computeExtent;
+  final int Function() computeLength;
+
+  @override
+  void recompute() {
+    final length = computeLength();
```

Comment:
I'm not sure this will provide any benefits in the case you're looking at where you want to animate elements in and out of the tree.

This will assume a constant height to all elements in the tree.  To use this with an animating size, you'll need a way of ensuring that the size and offset of widgets being built can change, while the size of nonbuilt widgets will not.

You'll also want to watch out for 'jumps' where recomputing the length of offscreen items has resulted in the contents of the scrollview popping up to the top of the screen.

It is relatively easy to write widget tests that cover both of these cases.

Reply:
The point of this class is that each item is given a custom height. Do you see a place where I am assuming a fixed height for all elements? 
`itemExtent(index)` and related methods are used to vend custom extents for each item.
With regard to animating, I have a test where the delegate animates the sizes where it is verified that the new layout matches what is expected. I can add some additional tests if you are concerned those tests aren't sufficient. Note that I already have more tests than the existing fixed extent list code this is ported from had in package:flutter. 
The ExtentDelegate is used to vend the size for the list so I expect there will be fewer edge cases where the size jumps than using a regular ListView where the total size is not known. As far as recomputing the total size, the intent is that as in the case of a ListView with a fixed extent, there isn't realy estimation of size of on-stage vs off-stage widgets. In both cases the delegate is used to determine offsets.